Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -12851,6 +12851,13 @@ void WebPageProxy::convertRectToMainFrameCoordinates(WebCore::FloatRect rect, st
     });
 }
 
+Awaitable<std::optional<WebCore::FloatRect>> WebPageProxy::convertRectToMainFrameCoordinates(WebCore::FloatRect rect, std::optional<WebCore::FrameIdentifier> frameID)
+{
+    co_return co_await AwaitableFromCompletionHandler<std::optional<WebCore::FloatRect>> { [protectedThis = Ref { *this }, rect, frameID] (auto completionHandler) {
```

Comment:
I wonder if there's any template deduction guide that can be written to make it so that the template type can be inferred in this line

Reply:
I don't think so because the argument is a lambda with an auto parameter.  The type needs to be somewhere.